libcextract/SymbolExternalizer.cpp

```diff
@@ -33,34 +33,50 @@
    FileEntry => FileID that we are sure to have modifications.  */
 #pragma GCC poison translateFile
 
-/****** Begin hack: used to get a pointer to a private member of a class. *****/
-struct ASTUnit_TopLevelDecls
+static std::vector<SourceRange>
+Get_Range_Of_Identifier_In_SrcRange(const SourceRange &range, const char *id)
 {
-  typedef std::vector<Decl*> ASTUnit::*type;
-  friend type Get(ASTUnit_TopLevelDecls);
-};
+  std::vector<SourceRange> ret = {};
+  StringRef string = PrettyPrint::Get_Source_Text(range);
+
+  /* Tokenize away the function-like macro stuff or expression, we only want
+     the identifier.  */
+  const char *token_vector = " ().,;+-*/^|&{}[]<>^&|\r\n\t";
```

Comment:
It would be better if this function is somehow refactored so we can reuse the logic in `Get_Range_Of_Identifier_In_SrcRange`.

But for a while I think it is enough to factor out the `token_vector` into a common variable, like a macro for example.